clang/lib/Analysis/FlowSensitive/Transfer.cpp

```diff
@@ -535,7 +535,19 @@ class TransferVisitor : public ConstStmtVisitor<TransferVisitor> {
         return;
 
       copyRecord(*LocSrc, *LocDst, Env);
-      Env.setStorageLocation(*S, *LocDst);
+
+      // If the expr is a glvalue, we can reasonably assume the operator is
+      // returning T& and thus we can assign it `LocDst`.
+      if (S->isGLValue()) {
+        Env.setStorageLocation(*S, *LocDst);
+      } else if (S->getType()->isRecordType() && S->isPRValue()) {
```

Comment:
```suggestion
      } else if (S->getType()->isRecordType()) {
```

`S->isPRValue()` is redundant here because `S->isPRValue() == !S->isGLValue()`.